### PR TITLE
Added better detection for protected binaries, fixes not loading into

### LINKED ISSUE #1397: mirrord fails to load to `go run .` on macOS
### Bug Description

```
> mirrord exec -c -n httpbin  -t pod/httpbin-0 go -- run .
⠧ mirrord cli starting
  ✓ ready to launch process
  ✓ layer extracted
  ✓ No Operator Detected
  ✓ agent pod created
  ✓ pod is ready
Starting server at port 8080
Error:
  × Internal proxy error: FirstConnectionTimeout
  ╰─▶ Couldn't accept connection before timeout
```
This happens because mirrord fails to load to Go, after debugging it I found out that the go binary is signed, and can't be loaded without patching. 
After adding it to `sip_binaries` it manages to load, but then fails on `fatal: bad g in signal handler`
debugging it further I found out the issue was actually `"local": ["/tmp", "/var/private", "/private"]` being accessed remotely.

### Steps to Reproduce

1.
```sh
go mod init example.com/app
```
2. Create app.go
```go
package main

import (
    "fmt"
    "net/http"
)

func main() {
    serverOne := &http.Server{
        Addr: ":8080",
        Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
            fmt.Printf("Req: %s %s\n", r.Host, r.URL.Path)
        }),
    }

    fmt.Printf("Starting server at port 8080\n")
    serverOne.ListenAndServe()
}
```
3. run using mirrord `mirrord exec -c -n httpbin  -t pod/httpbin-0 go -- run .`

### Backtrace

_No response_

### Relevant Logs

_No response_

### Your operating system and version

macOS

### Local process

go

### Local process version

go version go1.20 darwin/arm64

### Additional Info

_No response_

### PR DESCRIPTION
**Pull request by aviramha**
Go binary

Fixes #1397

### CODE REVIEW

File: changelog.d/1397#.fixed.md
Comment (t4lz) [inline changelog.d/1397#.fixed.md:1]: `#` after issue number

**Review by t4lz — APPROVED** [review]
Other than changelog file name - all good :)
Please just fix the filename

File: changelog.d/1397#.fixed.md
Comment (aviramha) [reply changelog.d/1397#.fixed.md:1]: It is on purpose - There's already same number issue and I found out that if I use `#` the link works and anyway the file will be removed on release

**Comment by aviramha** [comment]
Unblocked. Tested locally and looks good!

**Pull request merged**